### DISCUSSION TITLE
How to invoke a Method with tasty-query

### DISCUSSION THREAD

**Discussion by anjunar** [scalacenter/tasty-query#348] [Q&A] [answered]
When I have found a Method with tasty-query, how do I invoke it?

With Java you use Java Reflection, with Scala 2 you use Scala2 Reflection

How to invoke a Method with Scala 3 tasty-query?

**Comment by bishabosha** [comment, marked as answer]
tasty-query does not have any way to invoke methods. It only supports analysing the structure and semantics of definitions.

To invoke definitions, you would use Java reflection, and perhaps you could create a lookup method that converts a tasty-query Symbol to a java reflect Method.

**Reply by anjunar** [reply]
Thank you for the response =)

I have already solved it with Java Reflection... it would be nice, when tasty-query would support it out of the box

**Reply by sjrd** [reply]
It is clearly out of scope for tasty-query. Notably, it is designed to analyze arbitrary classpaths, not just the running application, so it doesn't even make sense to offer that as a feature of tasty-query.

**Reply by anjunar** [reply]
It is one piece that is missing in scala 3. 

To invoke methods, fields and constructors, or I haven't found it yet.

I tried Scala Macros, but it doesn't work.
